src/main/java/fi/aalto/cs/apluscourses/intellij/model/IntelliJCourseFactory.java

```diff
@@ -0,0 +1,33 @@
+package fi.aalto.cs.apluscourses.intellij.model;
+
+import com.intellij.openapi.project.Project;
+import fi.aalto.cs.apluscourses.model.Course;
+import fi.aalto.cs.apluscourses.model.CourseFactory;
+import fi.aalto.cs.apluscourses.model.Module;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+
+public class IntelliJCourseFactory implements CourseFactory {
```

Comment:
as it is capable of creating both a **course** and a **module** should it have a bit more descriptive (e.g. IntelliJCourseModuleFactory) or more generic (like IntelliJModelFactory)?

Reply:
Agreed! ModelFactory and its subclass IntelliJModelFactory seem more suitable names here.